vision/unit_tests/photo_model_test.py

```diff
@@ -0,0 +1,109 @@
+
+    # The list where all sightings of ODLCs will be stored
+    saved_odlcs: list[BoundingBox] = []
+
+    # The list of BoundingBoxes where all potential emergent objects will be stored
+    saved_humanoids: list[BoundingBox] = []
+
+    # Load in the json containing the camera data
+    image_parameters: dict[str, consts.CameraParameters] = pipe_utils.read_parameter_json(
+        camera_data_path
+    )
+
+    # Loop through all images in the json - if it hasn't been processed, process it
+    for image_path in image_parameters.keys():
+        print(image_path)
+        if image_path not in completed_images:
+            # Save the image path as completed so it isn't processed again
+            completed_images.append(image_path)
+
+            # Load the image to process
+            print("Loading image")
+            image: consts.Image = cv2.imread(image_path)
+
+            # Get the camera parameters from the loaded parameter file
+            print("Loading camera parameters")
+            camera_parameters: consts.CameraParameters = image_parameters[image_path]
+
+            # Append all discovered standard objects to the list of saved odlcs
+            print("Finding standard objects")
+            saved_odlcs += std_obj.find_standard_objects(image, camera_parameters, image_path)
+
+            # Append all discovered humanoids to the list of saved humanoids
+            print("Finding emergent objects")
+            saved_humanoids += emg_obj.find_humanoids(
+                emg_model, image, camera_parameters, image_path
+            )
+
+    print("All images have been processed.")
+    print(f"ODLC: {saved_odlcs}")
+    print(f"Humanoids: {saved_humanoids}")
+    # Sort and output the locations of all ODLCs
+    sorted_odlcs: list[list[BoundingBox]] = std_obj.sort_odlcs(bottle_info, saved_odlcs)
+    odlc_dict: consts.ODLCDict = std_obj.create_odlc_dict(sorted_odlcs)
+    print(odlc_dict)
+
+    j = 0
+    for odlc in saved_odlcs:
+        j += 1
+        print(f"Text: {odlc.get_attribute('text_color')} {odlc.get_attribute('text')}")
+        print(f"Shape: {odlc.get_attribute('shape_color')} {odlc.get_attribute('shape')}")
+        odlc_image: consts.Image = cv2.imread(odlc.get_attribute("image_path"))
+        odlc_crop: consts.Image = crop_image(odlc_image, odlc)
+
+        cv2.imwrite(f"odlc_object{j}.jpg", odlc_crop)
+
+    # Pick the emergent object and save the image cropped in on the emergent object
+    if len(saved_humanoids) > 0:
+        i = 0
+        for humanoid in saved_humanoids:
+            i += 1
+            emergent_image: consts.Image = cv2.imread(humanoid.get_attribute("image_path"))
+            emergent_crop: consts.Image = crop_image(emergent_image, humanoid)
+
+            cv2.imwrite(f"emergent_object{i}.jpg", emergent_crop)
+
+
+logging.basicConfig(filename="/dev/stdout", level=logging.INFO)
+flyover_pipeline("/home/alen/Documents/temp/photos/jun8/camera.json")
```

Comment:
This path will need to be changed for the actual Multirotor ground station, or you can generalize it using functions from the path module

Reply:
Changing it to the standard path for camera.json